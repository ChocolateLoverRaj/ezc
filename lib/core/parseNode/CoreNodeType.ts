enum CoreNodeType {
  FILE,
  DECLARE,
  FUNCTION,
  INTEGER_TYPE,
  GLOBAL_VARIABLE,
  STRING,
  FLOAT_TYPE,
  POINTER_TYPE,
  ARRAY_TYPE,
  STRUCT_TYPE,
  IDENTIFIER,
  NUMBER,
  VOID_TYPE,
  BLOCK,
  RETURN_INSTRUCTION,
  CALL_ASSIGNABLE,
  ASSIGNMENT_INSTRUCTION
}

export default CoreNodeType
