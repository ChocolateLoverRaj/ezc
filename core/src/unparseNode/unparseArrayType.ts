import CoreKeyWord from '../CoreKeyWord'
import CoreTokenType from '../CoreTokenType'
import CoreNodeDatas from '../parseNode/CoreNodeDatas'
import CoreNodeType from '../parseNode/CoreNodeType'
import UnparsedPartType from './UnparsedPartType'
import UnparseNode from './UnparseNode'

const unparseArrayType: UnparseNode<CoreNodeDatas[CoreNodeType.ARRAY_TYPE]> = ({
  length,
  itemsType
}) => [{
  type: UnparsedPartType.TOKEN,
  data: {
    type: {
      enum: CoreTokenType,
      id: CoreTokenType.KEY_WORD
    },
    data: {
      enum: CoreKeyWord,
      id: CoreKeyWord.OPEN_BRACKET
    }
  }
}, {
  type: UnparsedPartType.NODE,
  data: {
    type: {
      enum: CoreNodeType,
      id: CoreNodeType.NUMBER
    },
    data: {
      value: length,
      floatType: undefined
    }
  }
}, {
  type: UnparsedPartType.SPACE,
  data: undefined
}, {
  type: UnparsedPartType.TOKEN,
  data: {
    type: {
      enum: CoreTokenType,
      id: CoreTokenType.KEY_WORD
    },
    data: {
      enum: CoreKeyWord,
      id: CoreKeyWord.X
    }
  }
}, {
  type: UnparsedPartType.SPACE,
  data: undefined
}, {
  type: UnparsedPartType.NODE,
  data: itemsType
}, {
  type: UnparsedPartType.TOKEN,
  data: {
    type: {
      enum: CoreTokenType,
      id: CoreTokenType.KEY_WORD
    },
    data: {
      enum: CoreKeyWord,
      id: CoreKeyWord.CLOSE_BRACKET
    }
  }
}]

export default unparseArrayType
