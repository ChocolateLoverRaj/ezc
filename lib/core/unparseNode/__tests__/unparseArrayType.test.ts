import CoreNodeType from '../../parseNode/CoreNodeType'
import unparseArrayType from '../unparseArrayType'

test('[0 x i8]', () => {
  expect(unparseArrayType({
    length: 0,
    itemsType: {
      type: {
        enum: CoreNodeType,
        id: CoreNodeType.INTEGER_TYPE
      },
      data: 8
    }
  })).toMatchSnapshot()
})
