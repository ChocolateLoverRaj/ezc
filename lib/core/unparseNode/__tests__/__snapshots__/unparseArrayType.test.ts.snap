// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[0 x i8] 1`] = `
Array [
  Object {
    "data": Object {
      "data": Object {
        "data": 21,
        "enum": Object {
          "0": "PRIVATE",
          "1": "UNNAMED_ADDR",
          "10": "RETURN",
          "11": "EQUALS",
          "12": "GLOBAL",
          "13": "NO_CAPTURE",
          "14": "NO_ALIAS",
          "15": "COMMA",
          "16": "VOID",
          "17": "NO_FREE",
          "18": "CALL",
          "19": "OPEN_PARENTHESIS",
          "2": "CONSTANT",
          "20": "CLOSE_PARENTHESIS",
          "21": "OPEN_BRACKET",
          "22": "CLOSE_BRACKET",
          "23": "OPEN_CURLY_BRACKET",
          "24": "CLOSE_CURLY_BRACKET",
          "3": "ALIGN",
          "4": "DECLARE",
          "5": "DEFINE",
          "6": "PTR",
          "7": "GET_ELEMENT_PTR",
          "8": "INBOUNDS",
          "9": "X",
          "ALIGN": 3,
          "CALL": 18,
          "CLOSE_BRACKET": 22,
          "CLOSE_CURLY_BRACKET": 24,
          "CLOSE_PARENTHESIS": 20,
          "COMMA": 15,
          "CONSTANT": 2,
          "DECLARE": 4,
          "DEFINE": 5,
          "EQUALS": 11,
          "GET_ELEMENT_PTR": 7,
          "GLOBAL": 12,
          "INBOUNDS": 8,
          "NO_ALIAS": 14,
          "NO_CAPTURE": 13,
          "NO_FREE": 17,
          "OPEN_BRACKET": 21,
          "OPEN_CURLY_BRACKET": 23,
          "OPEN_PARENTHESIS": 19,
          "PRIVATE": 0,
          "PTR": 6,
          "RETURN": 10,
          "UNNAMED_ADDR": 1,
          "VOID": 16,
          "X": 9,
        },
      },
      "type": Object {
        "enum": Object {
          "0": "IDENTIFIER",
          "1": "NUMBER_LITERAL",
          "2": "INTEGER_TYPE",
          "3": "STRING_LITERAL",
          "4": "KEY_WORD",
          "IDENTIFIER": 0,
          "INTEGER_TYPE": 2,
          "KEY_WORD": 4,
          "NUMBER_LITERAL": 1,
          "STRING_LITERAL": 3,
        },
        "id": 4,
      },
    },
    "type": 0,
  },
  Object {
    "data": Object {
      "data": 0,
      "type": Object {
        "enum": Object {
          "0": "FILE",
          "1": "DECLARE",
          "10": "IDENTIFIER",
          "11": "NUMBER",
          "12": "VOID_TYPE",
          "13": "BLOCK",
          "14": "RETURN_INSTRUCTION",
          "15": "CALL_ASSIGNABLE",
          "16": "ASSIGNMENT_INSTRUCTION",
          "17": "INPUT",
          "18": "INPUT_FLAG",
          "2": "FUNCTION",
          "3": "INTEGER_TYPE",
          "4": "GLOBAL_VARIABLE",
          "5": "STRING",
          "6": "FLOAT_TYPE",
          "7": "POINTER_TYPE",
          "8": "ARRAY_TYPE",
          "9": "STRUCT_TYPE",
          "ARRAY_TYPE": 8,
          "ASSIGNMENT_INSTRUCTION": 16,
          "BLOCK": 13,
          "CALL_ASSIGNABLE": 15,
          "DECLARE": 1,
          "FILE": 0,
          "FLOAT_TYPE": 6,
          "FUNCTION": 2,
          "GLOBAL_VARIABLE": 4,
          "IDENTIFIER": 10,
          "INPUT": 17,
          "INPUT_FLAG": 18,
          "INTEGER_TYPE": 3,
          "NUMBER": 11,
          "POINTER_TYPE": 7,
          "RETURN_INSTRUCTION": 14,
          "STRING": 5,
          "STRUCT_TYPE": 9,
          "VOID_TYPE": 12,
        },
        "id": 11,
      },
    },
    "type": 2,
  },
  Object {
    "data": undefined,
    "type": 1,
  },
  Object {
    "data": Object {
      "data": Object {
        "data": 9,
        "enum": Object {
          "0": "PRIVATE",
          "1": "UNNAMED_ADDR",
          "10": "RETURN",
          "11": "EQUALS",
          "12": "GLOBAL",
          "13": "NO_CAPTURE",
          "14": "NO_ALIAS",
          "15": "COMMA",
          "16": "VOID",
          "17": "NO_FREE",
          "18": "CALL",
          "19": "OPEN_PARENTHESIS",
          "2": "CONSTANT",
          "20": "CLOSE_PARENTHESIS",
          "21": "OPEN_BRACKET",
          "22": "CLOSE_BRACKET",
          "23": "OPEN_CURLY_BRACKET",
          "24": "CLOSE_CURLY_BRACKET",
          "3": "ALIGN",
          "4": "DECLARE",
          "5": "DEFINE",
          "6": "PTR",
          "7": "GET_ELEMENT_PTR",
          "8": "INBOUNDS",
          "9": "X",
          "ALIGN": 3,
          "CALL": 18,
          "CLOSE_BRACKET": 22,
          "CLOSE_CURLY_BRACKET": 24,
          "CLOSE_PARENTHESIS": 20,
          "COMMA": 15,
          "CONSTANT": 2,
          "DECLARE": 4,
          "DEFINE": 5,
          "EQUALS": 11,
          "GET_ELEMENT_PTR": 7,
          "GLOBAL": 12,
          "INBOUNDS": 8,
          "NO_ALIAS": 14,
          "NO_CAPTURE": 13,
          "NO_FREE": 17,
          "OPEN_BRACKET": 21,
          "OPEN_CURLY_BRACKET": 23,
          "OPEN_PARENTHESIS": 19,
          "PRIVATE": 0,
          "PTR": 6,
          "RETURN": 10,
          "UNNAMED_ADDR": 1,
          "VOID": 16,
          "X": 9,
        },
      },
      "type": Object {
        "enum": Object {
          "0": "IDENTIFIER",
          "1": "NUMBER_LITERAL",
          "2": "INTEGER_TYPE",
          "3": "STRING_LITERAL",
          "4": "KEY_WORD",
          "IDENTIFIER": 0,
          "INTEGER_TYPE": 2,
          "KEY_WORD": 4,
          "NUMBER_LITERAL": 1,
          "STRING_LITERAL": 3,
        },
        "id": 4,
      },
    },
    "type": 0,
  },
  Object {
    "data": undefined,
    "type": 1,
  },
  Object {
    "data": Object {
      "data": 8,
      "type": Object {
        "enum": Object {
          "0": "FILE",
          "1": "DECLARE",
          "10": "IDENTIFIER",
          "11": "NUMBER",
          "12": "VOID_TYPE",
          "13": "BLOCK",
          "14": "RETURN_INSTRUCTION",
          "15": "CALL_ASSIGNABLE",
          "16": "ASSIGNMENT_INSTRUCTION",
          "17": "INPUT",
          "18": "INPUT_FLAG",
          "2": "FUNCTION",
          "3": "INTEGER_TYPE",
          "4": "GLOBAL_VARIABLE",
          "5": "STRING",
          "6": "FLOAT_TYPE",
          "7": "POINTER_TYPE",
          "8": "ARRAY_TYPE",
          "9": "STRUCT_TYPE",
          "ARRAY_TYPE": 8,
          "ASSIGNMENT_INSTRUCTION": 16,
          "BLOCK": 13,
          "CALL_ASSIGNABLE": 15,
          "DECLARE": 1,
          "FILE": 0,
          "FLOAT_TYPE": 6,
          "FUNCTION": 2,
          "GLOBAL_VARIABLE": 4,
          "IDENTIFIER": 10,
          "INPUT": 17,
          "INPUT_FLAG": 18,
          "INTEGER_TYPE": 3,
          "NUMBER": 11,
          "POINTER_TYPE": 7,
          "RETURN_INSTRUCTION": 14,
          "STRING": 5,
          "STRUCT_TYPE": 9,
          "VOID_TYPE": 12,
        },
        "id": 3,
      },
    },
    "type": 2,
  },
]
`;
